sources/models/src/lib.rs

```diff
@@ -179,10 +181,83 @@ struct ECSSettings {
     enable_spot_instance_draining: bool,
 }
 
+#[model]
+struct RegistryMirrors {
+    registry: SingleLineString,
+    endpoint: Vec<Url>,
+}
+
 // Image registry settings for the container runtimes.
 #[model]
 struct RegistrySettings {
-    mirrors: HashMap<SingleLineString, Vec<Url>>,
+    #[serde(deserialize_with = "deserialize_mirrors")]
+    mirrors: Vec<RegistryMirrors>,
+}
+
+// We want to allow two possible representations of registry mirrors. Either as a `Vec` of `RegistryMirrors`
+// or a `HashMap` of registry to endpoints.
+fn deserialize_mirrors<'de, D>(deserializer: D) -> Result<Option<Vec<RegistryMirrors>>, D::Error>
```

Comment:
I think it's useful to keep `models/src/lib.rs` quick to reference and understand, with only basic types, and using modules for more complex or shared types - can we move this to a module?